### PR TITLE
add support for ppc64le architecture

### PR DESCRIPTION
**Pull request by seth-priya**
Signed-off-by: Priya Seth <sethp@us.ibm.com>

### Description
add support for ppc64le architecture

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/1303

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by seth-priya** [comment]
The changes are based on https://github.com/opensearch-project/OpenSearch/pull/4001

The basic build (localDisto), precommit check pass, gradle-check on ppc64le fails with same issue as reported at https://github.com/opensearch-project/OpenSearch/issues/4173

**Comment by saratvemulapalli** [comment]
@seth-priya thank you for contributing the change. Is there a way to add a test for these changes? 
I know we do verify if the tars are generated for each architecture/distribution.

File: server/src/main/java/org/opensearch/bootstrap/SystemCallFilter.java [hunk 1]
```diff
@@ -260,6 +260,7 @@ static class Arch {
         m.put("amd64", new Arch(0xC000003E, 0x3FFFFFFF, 57, 58, 59, 322, 317));
         m.put("aarch64", new Arch(0xC00000B7, 0xFFFFFFFF, 1079, 1071, 221, 281, 277));
         m.put("s390x", new Arch(0x80000016, 0xFFFFFFFF, 2, 190, 11, 354, 348));
+        m.put("ppc64le", new Arch(0xC0000015, 0xFFFFFFFF, 2, 189, 11, 362, 358));
```
Comment (dbwiddis) [inline server/src/main/java/org/opensearch/bootstrap/SystemCallFilter.java:263]: (Preference) I would really like if at least some these values could be made constants.   I see above they correspond to [`AUDIT_ARCH_XXX`](AUDIT_ARCH_XXX), [`__NR_xxx`](https://github.com/torvalds/linux/blob/479174d402bcf60789106eedc4def3957c060bad/tools/testing/selftests/seccomp/seccomp_bpf.c#L130), etc.

**Review by dbwiddis — COMMENTED** [review]
Looks good... one fix needed (comments) and one suggestion (constants).

Agree with the question about tests -- I suspect we can at least test the enum determination by manipulating os.arch property.

**Review by dblock — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Review by saratvemulapalli — APPROVED** [review]
Thanks @seth-priya , looks like inherently gradle scripts are testing the expected distributions.

**Comment by saratvemulapalli** [comment]
@seth-priya could you update `CHANGELOG.md` and take care of conflicts. We should be good to merge.

**Comment by seth-priya** [comment]
thanks @saratvemulapalli for the feedback and review comments , agree on the test, addressed the conflict

**Comment by acarbonetto** [comment]
I know this is merged, but it seems to be causing fatal errors on the darwin-arm64-tar builds when running on m1 macs. 

```
» ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [runTask-0] fatal error in thread [main], exiting
»  java.lang.NoClassDefFoundError: org/opensearch/index/rankeval/RankEvalAction
»       at java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:?]
»       at java.lang.Class.privateGetDeclaredConstructors(Class.java:3373) ~[?:?]
»       at java.lang.Class.getConstructors(Class.java:2060) ~[?:?]
»       at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:765) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:731) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:533) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:195) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.node.Node.<init>(Node.java:427) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.node.Node.<init>(Node.java:354) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:404) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:180) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:171) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138) ~[opensearch-cli-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.cli.Command.main(Command.java:101) ~[opensearch-cli-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:137) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»       at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:103) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»  Caused by: java.lang.ClassNotFoundException: org.opensearch.index.rankeval.RankEvalAction
»       at java.net.URLClassLoader.findClass(URLClassLoader.java:445) ~[?:?]
»       at java.lang.ClassLoader.loadClass(ClassLoader.java:587) ~[?:?]
»       at java.net.FactoryURLClassLoader.loadClass(URLClassLoader.java:872) ~[?:?]
»       at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
»       ... 19 more
```

When I checkout the commit prior `git checkout 0f520f6ffe356f51584b8ed9dcec0f0ec24ef989` then my `./gradlew run` runs fine.

**Pull request merged**